index.ts

```diff
@@ -127,6 +128,27 @@ if (!oauthEnabled) {
   });
 }
 
+app.get("/qr/:type/:room?", async (req: express.Request, res: express.Response) => {
+  const options: QRCode.QRCodeToBufferOptions = {
+    width: 300,
+    color: {
+      dark: "000000ff",
+      light: "ffffff00"
+    }
+  }
+
+  let path: string;
+  if (req.params.type === "join") {
+    path = 'stimmung';
+  } else {
+    res.sendStatus(400);
+    return;
+  }
+
+  res.set('Content-Type', 'image/png');
+  res.send(await QRCode.toBuffer(`${process.env.OAUTH_REDIRECT_BASE || req.hostname}/${path}/${req.params.room || ''}`, options));
```

Comment:
`req.hostname` does not include protocol (and port) and thus results in an invalid URL if `OAUTH_REDIRECT_BASE` is not set.

Reply:
Oh. I guess I read the documentation wrong. My reasoning was that if OAuth is not being used, we would need a way to determine the URL. I actually forgot about the protocol (`req.secure ? 'https' : 'http'`) but thought the port would be included in `req.hostname`.